### PR TITLE
Implement InventoryMock#close()

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Implement InventoryMock#close()

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java [hunk 1]
```diff
@@ -526,8 +527,10 @@ public void clear()
 	@Override
 	public int close()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		int count = this.viewers.size();
+		System.out.println(viewers);
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java]: I don't think this was intended to be here :P

File: src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java [hunk 2]
```diff
@@ -526,8 +527,10 @@ public void clear()
...
+		Lists.newArrayList(this.viewers).forEach(HumanEntity::closeInventory);
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java:531]: Why is the list cloned?

File: src/test/java/be/seeseemelk/mockbukkit/inventory/InventoryMockTest.java [hunk 3]
```diff
@@ -595,4 +595,15 @@ void closeInventoryRemoveViewer()
...
 
+	@Test
+	void testClose()
+	{
+		Player player = server.addPlayer();
+		player.openInventory(inventory);
+		assertEquals(1, inventory.getViewers().size());
+		assertEquals(1, inventory.close());
+		assertEquals(0, inventory.getViewers().size());
+	}
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/InventoryMockTest.java:608]: There should be an assertion to ensure the player no longer has the inventory open

File: src/test/java/be/seeseemelk/mockbukkit/inventory/InventoryMockTest.java [hunk 4]
```diff
@@ -595,4 +595,15 @@ void closeInventoryRemoveViewer()
...
+
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/InventoryMockTest.java:609]:
```suggestion

**Review by Insprill — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java (hunk 2 above)
Comment (thelooter) [reply src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java:531]: This is the way Paper does it, so I kept it

File: src/test/java/be/seeseemelk/mockbukkit/inventory/InventoryMockTest.java [hunk 5]
```diff
@@ -595,4 +596,15 @@ void closeInventoryRemoveViewer()
...
 
+	@Test
+	void testClose()
+	{
+		Player player = server.addPlayer();
+		player.openInventory(inventory);
+		assertEquals(1, inventory.getViewers().size());
+		assertEquals(1, inventory.close());
+		assertEquals(0, inventory.getViewers().size());
+		assertNotSame(inventory,player.getOpenInventory().getTopInventory());
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/InventoryMockTest.java]: Missing space after the comma

**Review by Insprill — CHANGES_REQUESTED** [review]

**Review by Insprill — APPROVED** [review]
LGTM

**Pull request merged**